src/lightning_lite/lite.py

```diff
@@ -157,6 +159,10 @@ def setup(
         # Let accelerator/plugin wrap and connect the models and optimizers
         model, optimizers = self._strategy.setup_module_and_optimizers(model, list(optimizers))
         model = _LiteModule(model, self._precision_plugin, original_module=original_model)
+
+        # Update the _DeviceDtypeModuleMixin's device parameter
+        model._device = self.device if move_to_device else next(model.parameters()).device
```

Comment:
what's the issue with simply calling `.to` on the model for the respective dtype?

Reply:
No issue I guess, as at this point it would be a no-op with respect to the nn.Module. But I thought setting it this way would be clearer for us, as it shows that we are using `_DeviceDtypeModuleMixin` and need to work with the limitation. 

Would you prefer it the other way? A reader might come along and say why is it there, it seems redundant as the model has already moved in an earlier call

Reply:
No issue in that sense, just that the semantics of "move_to_device" does not imply the dtype is modified.

Reply:
This could avoid accessing protected API by doing

```suggestion
        model.to(self.device if move_to_device else next(model.parameters()).device)
```